into/backends/sql_csv.py

```diff
@@ -152,7 +154,9 @@ def append_csv_to_sql_table(tbl, csv, **kwargs):
     # move things to a temporary S3 bucket if we're using redshift and we
     # aren't already in S3
     if dialect == 'redshift' and not isinstance(csv, (S3(CSV), Temp(S3(CSV)))):
```

Comment:
`isinstance(...,  S3(CSV))` should be sufficient.  `isinstance` honors inheritance.